sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TableOutputResolver.scala

```diff
@@ -83,11 +82,13 @@ object TableOutputResolver {
       val matched = inputCols.filter(col => conf.resolver(col.name, expectedCol.name))
       val newColPath = colPath :+ expectedCol.name
       if (matched.isEmpty) {
-        addError(s"Cannot find data for output column '${newColPath.quoted}'")
-        None
+        throw QueryCompilationErrors.incompatibleDataToTableCannotFindDataError(
```

Comment:
Do you still need the `addError` parameter?

Reply:
IMHO, after this change it raises the error right on the line, so `addError` isn't necessary anymore here?

If I understood correctly the existing way of using `addError` collects multiple errors and outputs them at once, so it is difficult for the users to know exactly where the error occurs.